test/integration/http/ssl/index.js

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+export default function ({ getService }) {
+  const supertest = getService('supertest');
+
+  /* Node rejects unauthorized TLS requests by default.
+   * When testing https server, set this to false.
+   * https://github.com/visionmedia/supertest/issues/396
+   */
+  process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0';
```

Comment:
In general I think we should avoid disabling "reject unauthorized" behavior if possible, and since we are using a specific `ssl.certificate` when we start Kibana we should be able to use that certificate to indicate that servers using that certificate are "authorized". Probably with something like:

```js
export function KibanaSupertestProvider({ getService }) {
  const config = getService('config');
  const kibanaServerUrl = formatUrl(config.get('servers.kibana'));

  const agent = supertestAsPromised.agent();

  // maybe we should expose a separate config option for this?
  const kibanaServerCert = config.get('kbnTestServer.serverArgs')
    .filter(arg => arg.startsWith('--server.ssl.certificate'))
    .map(arg => arg.split('=').pop())
    .map(path => readFileSync(path))
    .shift()
  
  if (kibanaServerCert) {
    agent.ca(kibanaServerCert)
  }

  return agent(kibanaServerUrl)
}
```

That way we will also be ensuring that the server is using SSL and that it's using the right certificates. That's probably better, don't you think?

Reply:
Looks good. I agree. I didn't know about this option on supertest. TY!

Reply:
@spalger The problem is, the dev cert is self signed, and for some reason there's an error when using it: 
│ proc  [ftr]    └- ✖ fail: "kibana server with ssl handles requests using ssl"
   │ proc  [ftr]    │        Error: self signed certificate
   │ proc  [ftr]    │         at TLSSocket.<anonymous> (_tls_wrap.js:1105:38)
   │ proc  [ftr]    │         at TLSSocket._finishInit (_tls_wrap.js:639:8)
   │ proc  [ftr]    │         at TLSWrap.ssl.onhandshakedone (_tls_wrap.js:469:38)
[This](https://stackoverflow.com/questions/19816689/nodejs-supertest-testing-routes-with-certificates) might be bad advice, but it references the self signed cert issue. Any other ideas?

Reply:
Here's more context for what node is doing:
https://github.com/nodejs/node-v0.x-archive/pull/4023
```
tls, https: validate server certificates by default

This commit changes the default value of the rejectUnauthorized option from
false to true.

What that means is that tls.connect(), https.get() and https.request() will
reject invalid server certificates from now on, including self-signed
certificates.

There is an escape hatch: if you set the NODE_TLS_REJECT_UNAUTHORIZED
environment variable to the literal string "0", node.js reverts to its
old behavior.
```
Could it be that when setting `process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0';`, self-signed certs are still being accepted, as this comment implies:
https://github.com/nodejs/node-v0.x-archive/pull/4023/files#diff-cbfd1300d9c62cf7f325159582f45d66R27
>// disable strict server certificate validation by the client
>process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0';